includes/YOURLS/Config/InitDefaults.php

```diff
@@ -0,0 +1,137 @@
+<?php
+
+/**
+ * YOURLS defaut actions upon instantiating
+ *
+ * This class defines all the default actions to be performed when instantiating YOURLS. The idea
+ * is that this is easily tuneable depending on the scenario, namely when running YOURLS for
+ * unit tests.
+ *
+ * @see \YOURLS\Config\Init
+ */
+
+namespace YOURLS\Config;
+
+class InitDefaults {
```

Comment:
The idea of using an object instead of an array actually came from reading [something like this](https://github.com/jupeter/clean-code-php#function-arguments-2-or-fewer-ideally), which I find rather elegant.

Reply:
Haha, very good link! 😆 
That said, even if I strongly agree with that good practice, I found the example not really adapted. What is the interest of creating a class to make an array with all public attributes? It cost more space and CPU than an array and is much more complicated to handle...

Edit: ok, you can check the type of parameters, but without deep type check that comes with php7 it's useless IMHO...

Reply:
Interesting: replacing class `Config\InitDefaults` with an array within `Config\Init` and merging custom values ([code](https://gist.github.com/ozh/1b9f80aac785114e43a5e026f1b4166d)) eats 6% more memory on my computer...